shunting-yard-replicator/src/main/java/com/hotels/shunting/yard/replicator/exec/messaging/FilteringMessageReader.java

```diff
@@ -32,6 +33,12 @@ public FilteringMessageReader(MessageReader delegate, TableSelector tableSelecto
     this.tableSelector = tableSelector;
   }
 
+  /**
```

Comment:
remove this, this is not true. This class has nothing to do with the SQSMessageReader. This filtering will work for any implementation.
The `MessageReader` extends `Iterator` so hasNext will not be removed in the future. We just follow the iterator contract. See https://docs.oracle.com/javase/7/docs/api/java/util/Iterator.html.

Concerning the SQSMessageReader I understand return true is confusing but it does not break the contract. It doesn't say you cannot wait indefinetly for next to return a new value it just says: "return true if the iterator has more elements"